PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1

```diff
@@ -839,24 +839,45 @@ function Get-SddcDiagnosticInfo
     try { 
 		if ($ClusterName -eq ".")
 		{
-			$Cluster = Get-Cluster -Name $ClusterNodes[0].Name
-			$ClusterName = $Cluster.Name
+			foreach ($cn in $ClusterNodes)
+			{
+				$Cluster = Get-Cluster -Name $cn[0].Name -ErrorAction SilentlyContinue
+				
+				# if we cannot connect to cluster service will still have an access node this way
+				$AccessNode = $cn[0].Name
+				
+				if ($Cluster -eq $null)
+				{
+					continue;
+				}				
+				$ClusterName = $Cluster.Name
+				break;
+			}
 		}
 		else
 		{
 			$Cluster = Get-Cluster -Name $ClusterName
+			$AccessNode = $ClusterNodes[0].Name
 		}
 	}
     catch { Show-Error("Cluster could not be contacted. `nError="+$_.Exception.Message) }
-    if ($null -eq $Cluster) { Show-Error("Server is not in a cluster") }
-    $Cluster | Export-Clixml ($Path + "GetCluster.XML")
-
-    $ClusterName = $Cluster.Name + "." + $Cluster.Domain
-    $S2DEnabled = $Cluster.S2DEnabled
+	
+	if ($Cluster -ne $null)
+	{
+		$Cluster | Export-Clixml ($Path + "GetCluster.XML")
+		$ClusterName = $Cluster.Name + "." + $Cluster.Domain
+		$S2DEnabled = $Cluster.S2DEnabled
+		$ClusterDomain = $Cluster.Domain 
+	}
+	else
+	{
+		# We can only get here if -Nodelist was used, but cluster service isn't running
+		Write-Error "Cluster service was not running on any node, some information will be unavailable"
+		$ClusterName = "UNAVAILABLE";
```

Comment:
I'd be strongly inclined to use empty/null string v. a special value. Render as "UNAVAILABLE" when we're emitting it into reports, as needed. May need a more verbose comment though, such as indicating it is unavailable due to the cluster being down, etc. This is where we should export a recoverable indication of the state of the cluster so the summary report can repeat it. In the cluster-up case, we get GetCluster.XML.